File: benchmarks/string_concat/string_concat.js
const n = parseInt(process.argv[2]) || 100000;

const start = performance.now();

let result = "";
for (let i = 0; i < n; i++) {
    result += "a";
}

const elapsed = performance.now() - start;
console.error(`TIME_MS:${elapsed.toFixed(2)}`);
console.log(result.length);

File: benchmarks/string_concat/string_concat.rb
n = ARGV[0] ? ARGV[0].to_i : 100000

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)

result = ""
n.times do
  result += "a"
end

elapsed = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000
STDERR.puts "TIME_MS:#{format('%.2f', elapsed)}"
puts result.length
